enclave-tls/src/core/etls_common.c

```diff
@@ -0,0 +1,199 @@
+#include <stdlib.h>
+#ifndef SGX
+#include <dlfcn.h>
+#include <strings.h>
+#include <dirent.h>
+#endif
+#include <string.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include "enclave-tls/api.h"
+#include "enclave-tls/log.h"
+#include "internal/core.h"
+#include "err.h"
+
+#ifdef SGX
+#include "etls_t.h"
+
+extern void libcrypto_wrapper_nullcrypto_init(void);
+extern void libcrypto_wrapper_openssl_init(void);
+extern void libattester_null_init(void);
+extern void libverifier_null_init(void);
+extern void libattester_sgx_ecdsa_init(void);
+extern void libverifier_sgx_ecdsa_init(void);
+extern void libverifier_sgx_ecdsa_qve_init(void);
+extern void libattester_sgx_la_init(void);
+extern void libverifier_sgx_la_init(void);
+extern void libtls_wrapper_nulltls_init(void);
+extern void libtls_wrapper_openssl_init(void);
+#endif
+
+void etls_exit(void)
+{
+#ifndef SGX
+	exit(EXIT_FAILURE);
+#else
+	ocall_exit();
+#endif
+}
+
+enclave_tls_log_level_t etls_loglevel_getenv(const char *name)
+{
+	char *log_level_str = NULL;
+#ifdef SGX
+	size_t log_level_len = 32;
+	log_level_str = calloc(1, log_level_len);
+	memset(log_level_str, 0, log_level_len);
+	ocall_getenv(name, log_level_str, log_level_len);
+#else
+	log_level_str = getenv(name);
+#endif
+	if (log_level_str) {
+#ifdef SGX
+		if (!strcmp(log_level_str, "debug") || !strcmp(log_level_str, "DEBUG"))
+#else
+		if (!strcasecmp(log_level_str, "debug"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_DEBUG;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "info") || !strcmp(log_level_str, "INFO"))
+#else
+		else if (!strcasecmp(log_level_str, "info"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_INFO;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "warn") || !strcmp(log_level_str, "WARN"))
+#else
+		else if (!strcasecmp(log_level_str, "warn"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_WARN;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "error") || !strcmp(log_level_str, "ERROR"))
+#else
+		else if (!strcasecmp(log_level_str, "error"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_ERROR;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "fatal") || !strcmp(log_level_str, "FATAL"))
+#else
+		else if (!strcasecmp(log_level_str, "fatal"))
+#endif
+			return ENCLAVE_TLS_LOG_LEVEL_FATAL;
+#ifdef SGX
+		else if (!strcmp(log_level_str, "off") || !strcmp(log_level_str, "OFF"))
+#else
+		else if (!strcasecmp(log_level_str, "off"))
+#endif
```

Comment:
why not introducing a etls_strcmp() to handle this in sgx_utils.c? These codes looks painful.